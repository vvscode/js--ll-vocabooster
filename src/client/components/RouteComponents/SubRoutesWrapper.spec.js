import React from 'react';
import { shallow } from 'enzyme';
import Component from './SubRoutesWrapper';

describe('SubRoutesWrapper', () => {
  let props;
  let getComponent;

  beforeEach(() => {
    props = {
      header: null,
      footer: null,
      route: {
        path: '/test-path',
        routes: [],
      },
    };
    getComponent = () => shallow(<Component {...props} />);
  });

  it('renders header', () => {
    props.header = <div>Header</div>;
    expect(getComponent()).toMatchSnapshot();
  });

  it('renders footer', () => {
    props.footer = <div>Footer</div>;
    expect(getComponent()).toMatchSnapshot();
  });

  it('renders subrouted', () => {
    props.route.routes = [
      { path: '/test-path/p1', routes: [] },
      { path: '/test-path/p2', routes: [{ path: 'xxx', routes: [] }] },
    ];
    expect(getComponent()).toMatchSnapshot();
  });
});
