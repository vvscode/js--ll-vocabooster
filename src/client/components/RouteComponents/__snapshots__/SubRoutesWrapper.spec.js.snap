// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubRoutesWrapper renders footer 1`] = `
<div>
  <div>
    <Switch />
  </div>
  <div>
    Footer
  </div>
</div>
`;

exports[`SubRoutesWrapper renders header 1`] = `
<div>
  <div>
    Header
  </div>
  <div>
    <Switch />
  </div>
</div>
`;

exports[`SubRoutesWrapper renders subrouted 1`] = `
<div>
  <div>
    <Switch>
      <Route
        key="0"
        path="/test-path/p1"
        render={[Function]}
      />
      <Route
        key="1"
        path="/test-path/p2"
        render={[Function]}
      />
    </Switch>
  </div>
</div>
`;
